src/items.rs

```diff
@@ -73,15 +77,21 @@ impl Rewrite for ast::Local {
                 ),
                 shape,
                 false,
-            )?
+            )
+            .unknown_error()?
         };
         let let_kw_offset = result.len() - "let ".len();
 
         // 4 = "let ".len()
-        let pat_shape = shape.offset_left(4)?;
+        let pat_shape = shape
+            .offset_left(4)
+            .max_width_error(shape.width, self.span())?;
```

Comment:
On thing I've been wondering is whether we should use `shap.width` or `context.config.max_width()` for the max_width_error. Would love to get your thoughts on this.

Reply:
I believe `shape.width` provides more specific information, considering that rewrite tries to fit the formatted code into the given shape, while `context.config.max_width()` is set by the global configuration. Even if we want to expose only `max_width` to users in error messages (since `shape.width` might be confusing), in most cases, callers can retrieve `context.config.max_width()` from its rewrite context so converting `RewriteError` to `ErrorKind` would be possible.

However, if there isn't much opportunity to retry formatting based on the returned `RewriteError` with `shape.width` and `span`, it might be pointless to include shape.width, especially if this information is unused.

Reply:
I think it's fine to leave this as `shape.width` for now. We can always revisit this decision in the future if it turns out there's something better we should use instead of the width.